Add documentation to XkbToControl

wismill opened the pull request:
While some transformations are defined in the section “Interpreting the Control Modifier” of the XKB protocol, a lot are not. Documentation is scarce about the topic, so write down some context using the following sources:

- https://en.wikipedia.org/wiki/Caret_notation#Use_in_software
- https://www.vt100.net/shuford/terminal/dec_keyboards_news.txt
- https://misc.openbsd.narkive.com/NvSWf6ax/which-key-shortcuts-are-safe-to-bind-and-some-q-s-about-history-and-os-diffs-re-ctrl-4-means
- https://vt100.net/docs/vt220-rm/chapter3.html#T3-5

@whot @bluetech doc is really scarce, I hope you can give us some light!

Fixes #368

bluetech reviewed (approved):
Thanks, the doc LGTM.

Unfortunately I can't shed light on why the code deviates from the XKB spec. The `XkbToControl` function appears unchanged since at least [X11R6.1](https://www.x.org/archive/X11R6.1/) (released 1996) when XKB was added as an ISO 9995 implementation.

wismill commented on src/state.c:
```diff
@@ -894,7 +894,33 @@ should_do_ctrl_transformation(struct xkb_state *state, xkb_keycode_t kc)
+ * The `/` transformation seems to be defined for compatibility or convenience.
+ *
+ * [Interpreting the Control Modifier]: https://www.x.org/releases/current/doc/kbproto/xkbproto.html#Interpreting_the_Control_Modifier
+ * [carret notation]: https://en.wikipedia.org/wiki/Caret_notation
```
Good catch! I fixed the various occurrences. I should turn on the spell checking!

wismill commented:
Rebased and fixed typo.

> Unfortunately I can't shed light on why the code deviates from the XKB spec. The XkbToControl function appears unchanged since at least [X11R6.1](https://www.x.org/archive/X11R6.1/) (released 1996) when XKB was added as an ISO 9995 implementation.

@bluetech That is also what I found during my code speleology. There is maybe more info in XTerm, but I do not have time to go deeper in this topic. While this is very low priority, I think documenting these legacy parts of the code is still worth the effort, because someday nobody will have a clue and the code will bit rot.

whot commented:
Honestly - no idea myself here either, this goes much deeper than I ever had to look at it, so - LGTM, please merge.
